### PR TITLE
API changes to Azure Monitor exporter

### PR DESCRIPTION
**Pull request by srnagar**
This PR updates the Azure Monitor Exporter APIs to add support for the following:

- Include an option to configure the service API version when constructing the exporter instance

This change was recommended by the [arch board](https://github.com/Azure/azure-sdk/issues/1975).

### CODE REVIEW

**Comment by trask** [comment]
hey @srnagar, I think the preference is to not expose `instrumentationKey` or `endpoint` directly (only exposing `connectionString`)

**Comment by srnagar** [comment]
@trask updated the PR to remove the instrumentation key.

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/main/java/com/azure/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java [hunk 1]
```diff
@@ -213,12 +229,12 @@ MonitorExporterAsyncClient buildAsyncClient() {
      * implementation of OpenTelemetry {@link SpanExporter}.
      *
      * @return An instance of {@link AzureMonitorExporter}.
-     * @throws NullPointerException if the instrumentation key is not set.
+     * @throws NullPointerException if the instrumentation key or endpoint is not set.
```
Comment (trask) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/main/java/com/azure/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:232]:
```suggestion
     * @throws NullPointerException if the connection string is not set.
```

**Review by trask — APPROVED** [review]

**Review by joshfree — APPROVED** [review]

**Comment by joshfree** [comment]
LGTM

**Pull request merged**